source/fe/fe_q_bubbles.cc

```diff
@@ -0,0 +1,524 @@
+          Triangulation<dim, spacedim> tr;
+          GridGenerator::hyper_cube (tr, 0, 1);
+          tr.begin_active()->set_refine_flag(RefinementCase<dim>(ref_case));
+          tr.execute_coarsening_and_refinement();
+
+          DoFHandler<dim, spacedim> dh(tr);
+          dh.distribute_dofs(fe);
+
+          FEValues<dim, spacedim> fine (mapping, fe, q_fine,
+                                        update_quadrature_points
+                                        | update_JxW_values | update_values);
+
+          const unsigned int n_dofs = dh.n_dofs();
+
+          FullMatrix<double> fine_mass(n_dofs);
+          FullMatrix<double> coarse_rhs_matrix(n_dofs, dpc);
+
+          std::vector<std::vector<types::global_dof_index> > child_ldi
+          (nc, std::vector<types::global_dof_index>(fe.dofs_per_cell));
+
+          //now create the mass matrix and all the right_hand sides
+          unsigned int child_no = 0;
+          typename dealii::DoFHandler<dim, spacedim>::active_cell_iterator cell
+            = dh.begin_active();
+          for (; cell!=dh.end(); ++cell, ++child_no)
+            {
+              fine.reinit(cell);
+              cell->get_dof_indices(child_ldi[child_no]);
+
+              for (unsigned int q=0; q<nq; ++q)
+                for (unsigned int i=0; i<dpc; ++i)
+                  for (unsigned int j=0; j<dpc; ++j)
+                    {
+                      const unsigned int gdi=child_ldi[child_no][i];
+                      const unsigned int gdj=child_ldi[child_no][j];
+                      fine_mass(gdi, gdj)+=fine.shape_value(i,q)
+                                           *fine.shape_value(j,q)
+                                           *fine.JxW(q);
+                      coarse_rhs_matrix(gdi, j)
+                      +=fine.shape_value(i,q)
+                        *fe.shape_value(j, fine.quadrature_point(q))
+                        *fine.JxW(q);
+                    }
+            }
+
+          //now solve for all right-hand sides simultaneously
+          dealii::FullMatrix<double> solution (n_dofs, dpc);
+          fine_mass.gauss_jordan();
+          fine_mass.mmult(solution, coarse_rhs_matrix);
+
+          //and distribute to the fine cell matrices
+          for (unsigned int child_no=0; child_no<nc; ++child_no)
+            for (unsigned int i=0; i<dpc; ++i)
+              for (unsigned int j=0; j<dpc; ++j)
+                {
+                  const unsigned int gdi=child_ldi[child_no][i];
+                  //remove small entries
+                  if (std::fabs(solution(gdi, j)) > 1.e-12)
+                    matrices[ref_case-1][child_no](i,j)=solution(gdi, j);
+                }
+        }
+    }
+
+    template <>
+    inline
+    void
+    compute_embedding_matrices<1,2>(const FE_Q_Bubbles<1, 2> &fe,
+                                    std::vector<std::vector<FullMatrix<double> > > &matrices,
+                                    const bool isotropic_only)
+    {}
```

Comment:
What is the reason for forbidding the dim!=spacedim case? The embedding should be the result from the dim case, shouldn't it?